### PR TITLE
[bitnami/tomcat] initContainers and sidecars indent fix

### LINKED ISSUE #6501: [bitnami/tomcat] initContainers template generation is broken
**Which chart**:
bitnami/tomcat 9.2.7, problem exist since 9.2.4

Note: few minutes after posting this i found, that there is new 9.2.8 version, but problem persist
Note 2: after another few minutes i noticed, that very similar poblem is with sidecars, from version 9.2.4 too

**Describe the bug**
Broken template generation with initContainers in values

**To Reproduce**
1. create values yaml file with initContainers, for example init.yaml with this:
```
tomcatPassword: password123456789
initContainers:
    - name: init-test
      image: docker.io/bash:latest
      imagePullPolicy: "IfNotPresent"
      command:
          - sh
          - -c
          - ls -la
```

2. install tomcat chart, or test template, for example like this:
`helm template test-tomcat bitnami/tomcat -f init.yaml --debug > test-template.yaml`

3. result of previous command is:
```
install.go:173: [debug] Original chart version: ""
install.go:190: [debug] CHART PATH: /home/cache/.cache/helm/repository/tomcat-9.2.7.tgz

Error: YAML parse error on tomcat/templates/deployment.yaml: error converting YAML to JSON: yaml: line 49: did not find expected '-' indicator
helm.go:81: [debug] error converting YAML to JSON: yaml: line 49: did not find expected '-' indicator
YAML parse error on tomcat/templates/deployment.yaml
helm.sh/helm/v3/pkg/releaseutil.(*manifestFile).sort
        helm.sh/helm/v3/pkg/releaseutil/manifest_sorter.go:146
helm.sh/helm/v3/pkg/releaseutil.SortManifests
        helm.sh/helm/v3/pkg/releaseutil/manifest_sorter.go:106
helm.sh/helm/v3/pkg/action.(*Configuration).renderResources
        helm.sh/helm/v3/pkg/action/action.go:165
helm.sh/helm/v3/pkg/action.(*Install).Run
        helm.sh/helm/v3/pkg/action/install.go:245
main.runInstall
        helm.sh/helm/v3/cmd/helm/install.go:242
main.newTemplateCmd.func2
        helm.sh/helm/v3/cmd/helm/template.go:82
github.com/spf13/cobra.(*Command).execute
        github.com/spf13/cobra@v1.1.3/command.go:852
github.com/spf13/cobra.(*Command).ExecuteC
        github.com/spf13/cobra@v1.1.3/command.go:960
github.com/spf13/cobra.(*Command).Execute
        github.com/spf13/cobra@v1.1.3/command.go:897
main.main
        helm.sh/helm/v3/cmd/helm/helm.go:80
runtime.main
        runtime/proc.go:225
runtime.goexit
        runtime/asm_amd64.s:1371
```

3. part of generated yaml with initContainers:
```
      initContainers:- command:
        - sh
        - -c
        - ls -la
        image: docker.io/bash:latest
        imagePullPolicy: IfNotPresent
        name: init-test 
```

**Expected behavior**
Generate initContainers with correct indent.

**Version of Helm and Kubernetes**:

- Output of `helm version`:

```
version.BuildInfo{Version:"v3.6.0", GitCommit:"7f2df6467771a75f5646b7f12afb408590ed1755", GitTreeState:"clean", GoVersion:"go1.16.3"}
```

### PR DESCRIPTION
**Pull request by cache-sk**
**Description of the change**

Fix of indentations in tomcat chart

**Applicable issues**

  - fixes #6501

**Checklist** 
- [ ] Chart version bumped in `Chart.yaml` according to [semver](http://semver.org/).
- [ ] Variables are documented in the README.md
- [X] Title of the PR starts with chart name (e.g. [bitnami/<name_of_the_chart>])

NOTE: I did not know if i should bump version..

### CODE REVIEW

File: bitnami/tomcat/templates/_pod.tpl [hunk 1]
```diff
@@ -110,7 +110,7 @@ containers:
   {{- include "common.tplvalues.render" (dict "value" .Values.extraVolumeMounts "context" $) | nindent 2 }}
   {{- end }}
 {{- if .Values.sidecars }}
-{{- include "common.tplvalues.render" ( dict "value" .Values.sidecars "context" $) }}
+{{ include "common.tplvalues.render" ( dict "value" .Values.sidecars "context" $) }}
```
Comment (rafariossaa) [inline bitnami/tomcat/templates/_pod.tpl:113]: Hi,
  Could you provide some comment on this change ?

**Comment by rafariossaa** [comment]
Hi,
  Thanks for fixing this.
  Could to take a look to my comment and to the errors in the linter test ?.

File: bitnami/tomcat/templates/_pod.tpl [hunk 2]
```diff
@@ -43,7 +43,7 @@ initContainers:
     mountPath: /bitnami/tomcat
 {{- end }}
 {{- if .Values.initContainers }}
-{{- include "common.tplvalues.render" (dict "value" .Values.initContainers "context" $) }}
+{{- include "common.tplvalues.render" (dict "value" .Values.initContainers "context" $) | nindent 4 }}
```
Comment (migruiz4) [inline bitnami/tomcat/templates/_pod.tpl:46]: Hi @cache-sk,

Could you please apply the same fix used for the `.Values.sidecars` line?
```
 {{ include "common.tplvalues.render" (dict "value" .Values.initContainers "context" $) }}
```

The resulting indent using `nindent 4` is not technically bad:
```
      initContainers:
          - image: your-image
            imagePullPolicy: Always
            name: your-image-name
            ports:
            - containerPort: 1234
              name: portname
      containers:
      - name: tomcat
```

But for consistency reasons I would like the indent to be the same in the entire manifest.

File: bitnami/tomcat/templates/_pod.tpl (hunk 1 above)
Comment (cache-sk) [reply bitnami/tomcat/templates/_pod.tpl:113]: As i noted in #6501, sidecars have issue too, problem was that "-" removed previous whitespaces, include new line and template was generated like this:
```
        volumeMounts:
        - name: data
          mountPath: /bitnami/tomcat- image: bitnami/jmx-exporter:latest
        imagePullPolicy: IfNotPresent
        name: jmx-exporter
```

File: bitnami/tomcat/templates/_pod.tpl (hunk 2 above)
Comment (cache-sk) [reply bitnami/tomcat/templates/_pod.tpl]: Done.

**Comment by cache-sk** [comment]
Should i bump version to 9.2.9?

**Comment by migruiz4** [comment]
Yes please, a new patch version is required to merge this PR.

Thanks!

**Review by migruiz4 — APPROVED** [review]
Thank you very much for your contribution!

Your PR looks good to me.
As soon as the tests succeed I will merge your changes.

**Pull request merged**